lib/policies/load-balancing.js

```diff
@@ -209,20 +209,22 @@ DCAwareRoundRobinPolicy.prototype.newQueryPlan = function (keyspace, queryOption
     this.index = 0;
   }
   this._sliceNodesByDc();
+  // Use a local reference of hosts
+  var localHostsArray = this.localHostsArray;
```

Comment:
Just tried running the test with the previous version of the code and that seems to be exactly what is happening.